endstate_rew/system.py

```diff
@@ -15,20 +15,126 @@
     Simulation,
 )
 from openmmml import MLPotential
-from torch import randint
 from tqdm import tqdm
 
-from endstate_rew.constant import collision_rate, kBT, speed_unit, stepsize, temperature
+from endstate_rew.constant import (
+    collision_rate,
+    kBT,
+    speed_unit,
+    stepsize,
+    temperature,
+    zinc_systems,
+)
 
 forcefield = ForceField("openff_unconstrained-2.0.0.offxml")
 
 
-def generate_molecule(smiles: str, nr_of_conformations: int = 10) -> Molecule:
-    # generate a molecule using openff
-    molecule = Molecule.from_smiles(smiles, hydrogens_are_explicit=False)
-    molecule.generate_conformers(n_conformers=nr_of_conformations)
-    molecule.max_conf = nr_of_conformations
-    return molecule
+def generate_molecule(
+    forcefield: str,
+    name: str = "",
+    smiles: str = "",
+    base: str = "../data/hipen_data",
+    nr_of_conformations: int = 10,
+) -> Molecule:
+
+    # check which ff
+    if forcefield == "openff":
+
+        # if smiles string is not provided but zinc name is
+        if len(smiles) == 0 and len(name) > 0:
```

Comment:
I'd change the check to 
`if not smiles and name:`
A `str` with length 0 is `None`, so you don't have to check for `if len(string) == 0`, but can write `if not string`.
And the opposite is also true: you can write `if name` instead of `if len(name) > 0`